xc/device.ml

```diff
@@ -1743,43 +1743,70 @@ module Vusb = struct
     else
       []
 
-  let vusb_controller_plug ~xs ~domid ~driver ~driver_id =
-    if (Qemu.is_running ~xs domid)
-    && not (List.mem driver_id (qom_list ~xs ~domid))then
-      qmp_send_cmd domid Qmp.(Device_add Device.({driver; device=USB { USB.id=driver_id; params=None }})) |> ignore
 
-  let vusb_plug ~xs ~privileged ~domid ~id ~hostbus ~hostport ~version =
+  let vusb_plug ~xs ~privileged ~domid ~id ~hostbus ~hostport ~version ~speed =
     debug "vusb_plug: plug VUSB device %s" id;
-    let get_bus v =
-      if String.startswith "1" v then
-        "usb-bus.0"
-      else begin
-        (* Here plug usb controller according to the usb version*)
-        let usb_controller_driver = "usb-ehci" in
-        let driver_id = "ehci" in
-        vusb_controller_plug ~xs ~domid ~driver:usb_controller_driver ~driver_id;
-        Printf.sprintf "%s.0" driver_id;
-      end
+    let get_bus () =
+      let vusb_controller_plug ~xs ~domid ~driver ~driver_id =
```

Comment:
You already have `~xs ~domid` etc. from the outer scope, do you need to repeat them here?